monkey/monkey_island/cc/services/agent_plugin_service/agent_plugin_service.py

```diff
@@ -71,22 +74,60 @@ def install_agent_plugin_archive(self, agent_plugin_archive: bytes):
                     operating_system=operating_system, agent_plugin=agent_plugin
                 )
 
-    def get_available_plugins(self, force_refresh: bool) -> AgentPluginRepositoryIndex:
+    def install_agent_plugin_from_repository(
+        self, plugin_type: AgentPluginType, plugin_name: PluginName, plugin_version: PluginVersion
+    ):
+        plugin_metadata = self._find_plugin_in_repository(plugin_type, plugin_name, plugin_version)
+
+        plugin_file_path = plugin_metadata.resource_path
+        plugin_download_url = self._get_file_download_url(
+            file_path_in_repository=str(plugin_file_path)
+        )
```

Comment:
NIT: You could make `_get_file_download_url()` accept the metadata object and call `plugin_metadata.resource_path`. This method would be shorter and `_get_file_download_url()` would have more purpose.